### PR TITLE
consolidate helm settings

### PR DESCRIPTION
**Pull request by mlholland**
# Description

Adds a feature-gated partial to gloo-OS's settings template, which will be defined in gloo-E to inject enterprise-only settings into gloo without surfacing enterprise-only structs in OS. Also added an extauth feature to ensure that settings.spec.extauth is only defined once (since it's defined differently in enterprise)

Settings metadata was also modified slightly to match what's expected in gloo-E.

# Context

https://github.com/solo-io/gloo/issues/3470.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3470

### CODE REVIEW

File: install/helm/gloo/templates/18-settings.yaml [hunk 1]
```diff
@@ -97,12 +98,19 @@ spec:
...
 {{- if .Values.global.extensions.extAuth }}
   extauth:
 {{- toYaml .Values.global.extensions.extAuth | nindent 4 }}
 {{- end }}
 {{- end }}
+{{- end -}}
+
+{{- if .Values.global.addEnterpriseSettings }}
+{{- include "glooe.enterprisespecs"  . }}
+{{- end -}}
```
Comment (kdorosh) [inline install/helm/gloo/templates/18-settings.yaml:113]: I would change these names to be more generic, similar to https://github.com/solo-io/gloo/blob/3ffa8c2a37ed7e12d4ce97e6aa4de407d85dcef9/install/helm/gloo/templates/7-gateway-proxy-deployment.yaml#L115-L118

Comment (mlholland) [reply install/helm/gloo/templates/18-settings.yaml]: changing to 'extraSpecs'

File: install/helm/gloo/generate/values.go [hunk 2]
```diff
@@ -26,13 +26,15 @@ type Config struct {
...
+	GlooStats         Stats            `json:"glooStats,omitempty" desc:"Config used as the default values for Prometheus stats published from Gloo Edge pods. Can be overridden by individual deployments"`
+	GlooMtls          Mtls             `json:"glooMtls,omitempty" desc:"Config used to enable internal mtls authentication"`
+	IstioSDS          IstioSDS         `json:"istioSDS,omitempty" desc:"Config used for installing Gloo Edge with Istio SDS cert rotation features to facilitate Istio mTLS"`
+	IstioIntegration  IstioIntegration `json:"istioIntegration,omitempty" desc:"Configs user to manage Gloo pod visibility for Istio's' automatic discovery and sidecar injection."`
+	ExtraSpecs        bool             `json:"extraSpecs" desc:"Add additional specs to include in the settings manifest, as defined by a helm partial. Defaults to false in open source, and true in enterprise."`
+	ExtauthCustomYaml bool             `json:"extauthCustomYaml" desc:"Inject whatever yaml exists in .Values.global.extensions.extAuth into settings.spec.extauth, instead of structured yaml (which is enterprise only). Defaults to true in open source, and false in enterprise"`
```
Comment (kdorosh) [inline install/helm/gloo/generate/values.go:37]: let's add omitempty to every single value in this generate script

**Comment by kdorosh** [comment]
superseded by https://github.com/solo-io/gloo/pull/4412

**Pull request closed without merging**